src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java

```diff
@@ -158,32 +161,33 @@ public void tickServer()
 
 		if(!isRSDisabled()&&!tanks[0].getFluid().isEmpty())
 		{
-			GeneratorFuel recipe = recipeGetter.apply(level, tanks[0].getFluid().getFluid());
-			if(recipe!=null)
+			int output = IEServerConfig.MACHINES.dieselGen_output.get();
+			List<IEnergyStorage> presentOutputs = outputs.stream()
+					.map(CapabilityReference::getNullable)
+					.filter(Objects::nonNull)
+					.collect(Collectors.toList());
+			if(!presentOutputs.isEmpty()&&EnergyHelper.distributeFlux(presentOutputs, output, false) < output)
 			{
-				int burnTime = recipe.getBurnTime();
-				int fluidConsumed = FluidAttributes.BUCKET_VOLUME/burnTime;
-				int output = IEServerConfig.MACHINES.dieselGen_output.get();
-				List<IEnergyStorage> presentOutputs = outputs.stream()
-						.map(CapabilityReference::getNullable)
-						.filter(Objects::nonNull)
-						.collect(Collectors.toList());
-				if(!presentOutputs.isEmpty()&&
-						tanks[0].getFluidAmount() >= fluidConsumed&&
-						// Sort receivers by lowest input
-						EnergyHelper.distributeFlux(presentOutputs, output, false) < output)
+				consumeTick--;
+			}
+			if (consumeTick == 0) //Consume 10*tamount every 10t to allow for 1/10th mB amounts
```

Comment:
Typo in the comment? "tamount"

Reply:
Ohhh - no, that's an abbreviation for "tick-amount"
Basically, "consume ten times the amount per tick every ten ticks"